SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +656,237 @@ private void DisplayErrorMessage(string message)
                     MessageBoxButton.OK, MessageBoxImage.Error);
         }
 
+        private void LoginButton_Clicked(object sender, RoutedEventArgs e)
+        {
+            LabelEmptyFieldsError.Visibility = Visibility.Collapsed;
+            LabelUsernameError.Visibility = Visibility.Collapsed;
+            LabelPasswordError.Visibility = Visibility.Collapsed;
+
+            var users = userLoginData.GetAllData();
+            var usernameExists = false;
+            string pswHash = null;
+            string pswSalt = null;
+            string username = null;
+            string userID = null;            
+
+            foreach (var name in users)
+            {
+                var checkUsernameExists = (name.Username == UsernameTextBox.Text);
+                if (checkUsernameExists)
```

Comment:
Why is this boolean outside the if? It doesn't seem to be used anywhere else.

Reply:
Because I tested this thing when it wasn't working (appears for non-related thing) and left that. My memory is trash